### PR TITLE
initial CRC support

### PR DESCRIPTION
**Pull request by yjh0502**
I'm complete newcomer on microcontroller & stm32, so please feel free to comment.

 - Tested on `stm32f103c8`
 - I'm not sure how it interacts with other peripherals. For example it seems that SPI protocol requires CRC calculation. Does stm32 SPI uses CRC peripherals for CRC calculation of there's another CRC hardware dedicated to SPI CRC?
 - I'm not sure if I should use `split` or `constrain`.
 - small additional diff on `prelude` caused by `rustfmt`.

### CODE REVIEW

File: src/crc.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    pub dr: DR,
+    pub cr: CR,
+}
+
+/// Opaque DR register
+pub struct DR {
+    _0: (),
+}
+
+#[allow(dead_code)]
```
Comment (TheZoq2) [inline src/crc.rs:33]: This probably shouldn't be needed

Comment (burrbull) [reply src/crc.rs:33]: Really, why not just:
```
    let mut crc = Crc::new(p.CRC, &mut rcc.ahb);

    crc.reset();
    crc.write(0x12345678);

    let val = crc.read();
```

Comment (yjh0502) [reply src/crc.rs]: Fixed, could you review it again?

File: src/crc.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    crc: CRC,
+}
+
+impl Crc {
+    pub fn read(&self) -> u32 {
+        self.crc.dr.read().bits()
+    }
+
+    pub fn write(&mut self, val: u32) {
+        unsafe { self.crc.dr.write(|w| w.bits(val)) }
```
Comment (burrbull) [inline src/crc.rs:30]: This `unsafe` is unneeded:
https://docs.rs/stm32f1/0.11.0/stm32f1/stm32f103/crc/dr/struct.DR_W.html

Comment (yjh0502) [reply src/crc.rs]: Fixed, thanks!

**Comment by TheZoq2** [comment]
Looks good to me, thanks!

**Comment by TheZoq2** [comment]
Actually, would you mind adding an entry to the `Changelog.md` before we merge?

**Comment by yjh0502** [comment]
@TheZoq2 I updated `CHANGELOG.md` :)

**Comment by TheZoq2** [comment]
Lovely, thanks!

**Pull request merged**